Supermatter CO2 balance, helium and antinoblium gas supermatter interactions (sane this time).

Pickle-Coding opened the pull request:
CO2 heat penalty reduced to 2 from 6. Reduces bonus power transmission of CO2 to 1.5 from 2.

Powerloss inhibition is now based on the composition of gases in a list of gases mapped with their powerloss inhibition values instead of just co2 composition. The list includes co2, helium, and antinoblium.

Helium interaction with supermatter. Inhibits powerloss, has heat penalty of 6, gasmix power ratio of 1, and power transmission of 5. Gets converted to tritium when the SM's energy is above 5GeV.

Antinoblium interaction with supermatter. Inhibits powerloss, has heat penalty of 23, gasmix power ratio of 1, and power transmission of 40.

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Reduces CO2's heat penalty to 2, from 6. CO2's bonus transmission to zaps have been reduced. Adds helium interaction. Helium gets converted to tritium when SM is above 5GeV, and also reduces powerloss like CO2. Adds antinoblium interaction. Antinoblium has a heat penalty of 23, and inhibits powerloss like CO2, but has a power transmission bonus of 40.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
### CO2
CO2 having a heat penalty of 6 means it is much harder to keep it at high composition and low temperature, both being a requirement to achieve high power without delamming, which is what CO2 is about. While it should be challenging to achieve high power levels, the high heat penalty restrained the ranges of power most engines could achieve without delamming, which made CO2's gimmick look a lot more like an ordinary high heat penalty gas than a powerloss inhibitor gas. Lowering the heat penalty should let engineers to have a taste of increasing the SM's power levels with a better setup, but should still be a challenging task if they want to get to the very high power levels.

### Helium
Enabling helium to get converted to tritium under dangerous conditions should give the gas a purpose outside of the HFR. Although a gas's entire purpose outside of the HFR is to get converted to a useful gas is probably not the best or most exciting purpose, at least it gives people something to do with it other than selling it after extracting from HFR.

### Antinoblium
People said they could do tritium engines, so I will introduce antinoblium as the gas probably no engine would be able to maintain at 100% composition.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
balance: Reduces CO2 supermatter heat penalty to 2, from 6.
balance: Reduces the additional zap power transmission gain from CO2.
add: Adds helium gas interaction for the supermatter. It inhibits powerloss, and gets converted to tritium when the SM has a power above 5GeV. Heat penalty of 6.
add: Adds antinoblium gas interaction for the supermatter. It will probably destroy your engine.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Pickle-Coding committed:
Supermatter gas interactions.

CO2 heat penalty reduced to 2 from 6. CO2 no longer directly increases power transmission to zaps.

Powerloss inhibition is now based on the composition of gases in a list of gases mapped with their powerloss inhibition values instead of just co2 composition. The list includes co2, helium, and antinoblium.

Helium interaction with supermatter. Inhibits powerloss, has heat penalty of 6, gasmix power ratio of 1, and power transmission of 5. Gets converted to tritium when the SM's energy is above 5GeV.

Antinoblium interaction with supermatter. Inhibits powerloss, has heat penalty of 23, gasmix power ratio of 1, and power transmission of 40.

Pickle-Coding commented:
I'd like to hear Ghilker's opinions on the CO2 changes, and what CO2 should be for the supermatter engine.

Pickle-Coding commented:
Why do integration tests hate my PRs?

Ghilker commented on code/__DEFINES/supermatter.dm:
```diff
@@ -2,7 +2,7 @@
 #define PLUOXIUM_HEAT_PENALTY -0.5 //Better then co2, worse then n2
 #define TRITIUM_HEAT_PENALTY 10
-#define CO2_HEAT_PENALTY 6
+#define CO2_HEAT_PENALTY 2
```
Thinking about it overnight, this might lead to some interesting gameplay options. The only thing is the loss of co2 power transmittance increase, you could have it multiplied by a quarter so that co2 still retains some interesting quirks that make it much better than an average N2 SM

Ghilker commented on code/modules/power/supermatter/supermatter_process.dm:
```diff
@@ -266,11 +280,10 @@
 		playsound(src, 'sound/weapons/emitter2.ogg', 70, TRUE)
 		var/power_multiplier = max(0, (1 + (power_transmission_bonus / (10 - (gas_comp[/datum/gas/bz] * BZ_RADIOACTIVITY_MODIFIER)))) * freonbonus)// RadModBZ(500%)
 		var/pressure_multiplier = max((1 / ((env.return_pressure() ** pressure_bonus_curve_angle) + 1) * pressure_bonus_derived_steepness) + pressure_bonus_derived_constant, 1)
-		var/co2_power_increase = max(gas_comp[/datum/gas/carbon_dioxide] * 2, 1)
```
you could have this be `max(1 + gas_comp[/datum/gas/carbon_dioxide] * 0.5, 1)` 
this way it would be a multiplier from 1 to a theoretical maximum of 1.5

Pickle-Coding replied:
Would there be a reason to keep the max check, since that would always be at least 1 assuming the composition wouldn't be negative?

Pickle-Coding committed:
CO2

Brings back CO2 power increase at a reduced rate.

Pickle-Coding committed:
Removes define.

Removes helium gasmix supermatter define due to being useless and misleading.

Pickle-Coding committed:
Oops

Lets assume tritium is T2, based on hydrogen conversion reactions. Halves tritium output from helium to tritium interaction via supermatter.

LemonInTheDark commented:
> CO2 no longer directly increases power transmission to zaps.

Why?

LemonInTheDark commented on code/modules/power/supermatter/supermatter_process.dm:
```diff
@@ -214,6 +219,15 @@
 			removed.gases[/datum/gas/pluoxium][MOLES] += consumed_carbon_dioxide * 0.25
 
+	//Very high energy radiaton will convert helium into tritium.
+	if(gas_comp[/datum/gas/helium] && power > POWER_PENALTY_THRESHOLD)
+		var/helium_pp = env.return_pressure() * gas_comp[/datum/gas/helium]
+		var/consumed_helium = clamp(((helium_pp - HELIUM_CONSUMPTION_PP) / (helium_pp + HELIUM_PRESSURE_SCALING)) * gasmix_power_ratio * clamp((power - POWER_PENALTY_THRESHOLD) / POWER_PENALTY_THRESHOLD, 0, 1), HELIUM_CONSUMPTION_RATIO_MIN, HELIUM_CONSUMPTION_RATIO_MAX)
+		consumed_helium = min(consumed_helium * gas_comp[/datum/gas/helium] * combined_gas, removed.gases[/datum/gas/helium][MOLES])
+		if(consumed_helium)
+			removed.gases[/datum/gas/helium][MOLES] -= consumed_helium
+			removed.gases[/datum/gas/tritium][MOLES] += consumed_helium * 0.5
```
Do you think this is something people will actually use?

Pickle-Coding replied:
Probably in longer rounds when managing gases becomes more important, but in shorter rounds it's not that useful.

LemonInTheDark replied:
why should we add it then?

Pickle-Coding replied:
It allows a way to quickly but dangerously recycle helium back into tritium after using tritium in HFR. Might be interesting for people that want to try and conserve their gases and not waste them.

Pickle-Coding commented:
It does but I added it back at a reduced rate in a later commit, and forgot to change that in whatever part of the description you got that from.

I reduced it because it will be easier to achieve higher power levels with CO2

LemonInTheDark commented on code/__DEFINES/supermatter.dm:
```diff
@@ -24,14 +26,16 @@
 #define PROTO_NITRATE_TRANSMIT_MODIFIER 15
 #define ZAUKER_TRANSMIT_MODIFIER 20
+#define HELIUM_TRANSMIT_MODIFIER 5
+#define ANTINOBLIUM_TRANSMIT_MODIFIER 40
```
brotherrrrrrr

Pickle-Coding replied:
It's harder than tritium, so it might as well have a good reward. Would a power transmission of 40 break power balance?

LemonInTheDark replied:
I'm not convinced there's a point to making harder tritium, why is that something we need?

Pickle-Coding replied:
I use to be unsure on whether pure tritium engines were possible or not, and since it was basically the hardest gas, I got kinda disappointed when I found out how to run pure tritium engines, knowing there isn't really anything much harder to do.

I added the antinoblium interaction because I want to keep things fresh, and would be a fun challenge for anyone wanting to do something harder than tritium.

LemonInTheDark reviewed (changes requested):
I don't believe that people have ran trit engines, and I'm also not sure I like the idea of continually escalating difficulty until people give up

Pickle-Coding commented:
Changed description.

Pickle-Coding commented:
Pretty sure people have ran pure tritium engines, and one claimed to reach almost below 50K with pure tritium, so there's plenty of room for difficulty scaling.

LemonInTheDark commented:
> Pretty sure people have ran pure tritium engines, and one claimed to reach almost below 50K with pure tritium, so there's plenty of room for difficulty scaling.

When you say ran, what context is that in?

Pickle-Coding commented:
I've seen a few tritium engine builds, though they were on Skyrat. The 50K one was in a private server, so that's more of a theoretical engine.

LemonInTheDark commented:
When you say on skyrat, do you mean like 4 hours in or something?
I'm fine with trit being possible if you strain things to the limit, or close to it.

Pickle-Coding commented:
I forgot how long the round was, though I don't think it needs to take 4 hours. It's probably quicker to build now that thermomachines are back to 1 node.

Maybe 90-180 minutes if you are building it by yourself. Could be shorter if there is a competant engineering and atmospherics team working on the project together.
